src/instructlab/model/list.py

```diff
@@ -80,72 +87,36 @@ def print_table(headers: list[str], data: list[list[str]]):
-        "%Y-%m-%d %H:%M:%S", time.localtime(modification_time)
-    )
-    return [entry.name, modification_time_readable, f"{f_size:.1f} {magnitude}"]
+def _analyze_gguf(entry: Path) -> ModelInfo:
+    stat = entry.stat(follow_symlinks=False)
+    return get_model_info(entry.name, stat.st_mtime, stat.st_size)
 
 
 def _analyze_dir(
     entry: Path, list_checkpoints: bool, directory: str
-) -> AnalyzeResultDir:
-    actual_model_name = ""
-    all_files_sizes = 0
-    add_model = False
-    models = []
-    # walk entire dir.
-    for root, _, files in os.walk(entry.as_posix()):
-        normalized_path = os.path.normpath(root)
-        # Split the path into its components
-        parts = normalized_path.split(os.sep)
-        # Get the last two parts and join them back into a path
-        last_two_parts = os.path.join(parts[-2], parts[-1])
-        # if this is a dir it could be:
-        # top level repo dir `instructlab/`
-        # top level model dir `instructlab/granite-7b-lab`
-        # checkpoint top level dir `step-19`
-        # any lower level dir: `instructlab/granite-7b-lab/.huggingface/download.....`
-        # so, check if model is valid Safetensor, GGUF, or list it regardless w/ `--list-checkpoints`
-        # if --list-checkpoints is specified, we will list all checkpoints in the checkpoints dir regardless of the validity
-        if is_model_safetensors(Path(normalized_path)) or is_model_gguf(
-            Path(normalized_path)
-        ):
-            actual_model_name = last_two_parts
-            all_files_sizes = 0
-            add_model = True
-        else:
-            if list_checkpoints and directory is DEFAULTS.CHECKPOINTS_DIR:
-                logging.debug("Including model regardless of model validity")
-            else:
-                continue
-        for f in files:
-            # stat each file in the dir, add the size in Bytes, then convert to proper magnitude
-            full_file = os.path.join(root, f)
-            stat = Path(full_file).stat(follow_symlinks=False)
-            all_files_sizes += stat.st_size
-        all_files_sizes, magnitude = convert_bytes_to_proper_mag(all_files_sizes)
-        if add_model:
-            # add to table
-            modification_time = os.path.getmtime(entry.absolute())
-            modification_time_readable = time.strftime(
-                "%Y-%m-%d %H:%M:%S", time.localtime(modification_time)
-            )
-            models.append(
-                [
-                    actual_model_name,
-                    modification_time_readable,
-                    f"{all_files_sizes:.1f} {magnitude}",
-                ]
-            )
-    return models
+) -> Generator[ModelInfo, None, None]:
+    list_all = list_checkpoints and directory == DEFAULTS.CHECKPOINTS_DIR
+    # TODO: switch to Path.walk after python3.12 bump
+    for root_, _, files in os.walk(entry.as_posix()):
+        root = Path(root_)
+        if not (list_all or is_model_safetensors(root)):
```

Comment:
methinks now I should have is_model_gguf here back. Thought it's useless but I misread the intent. The intent I think is when pointing at the top level repo dir

Reply:
Hm, maybe not. `root` is NEVER a file here. So it will never be a gguf. Perhaps the intent (but not implementation) of `model list` was to drill down the directory and report on each found gguf file? But then one would have to check is_model_* on `files`, not on `root` (as done in the original code and with the refactor).

@cdoern what was the intent of `is_model_gguf` check here?

Reply:
@jaideepr97 FYI I think you are fixing this bug mentioned above in your PR